src/RA_Defs.h

```diff
@@ -207,69 +316,28 @@ class ResizeContent
     };
 
 public:
-    HWND hwnd;
-    POINT pLT;
-    POINT pRB;
-    AlignType nAlignType;
-    int nDistanceX;
-    int nDistanceY;
-    bool bResize;
-
-    ResizeContent(HWND parentHwnd, HWND contentHwnd, AlignType newAlignType, bool isResize)
-    {
-        hwnd = contentHwnd;
-        nAlignType = newAlignType;
-        bResize = isResize;
-
-        RARect rect;
-        GetWindowRect(hwnd, &rect);
-
-        pLT.x = rect.left;	pLT.y = rect.top;
-        pRB.x = rect.right; pRB.y = rect.bottom;
-
-        ScreenToClient(parentHwnd, &pLT);
-        ScreenToClient(parentHwnd, &pRB);
-
-        GetWindowRect(parentHwnd, &rect);
-        nDistanceX = rect.Width() - pLT.x;
-        nDistanceY = rect.Height() - pLT.y;
+    inline constexpr ResizeContent() noexcept = default;
+    // This one can't be constexpr
+    ResizeContent(_In_ HWND parentHwnd, _In_ HWND contentHwnd, _In_ AlignType newAlignType,
+        _In_ bool isResize) noexcept;
+    // TODO: make this defaulted once the RAII wrapper for HWND gets in
+    ~ResizeContent() noexcept = default;
+    ResizeContent(const ResizeContent&) = delete;
+    ResizeContent& operator=(const ResizeContent&) = delete;
+    inline constexpr ResizeContent(ResizeContent&&) noexcept = default;
+    _NODISCARD inline constexpr ResizeContent& operator=(ResizeContent&&) noexcept = default;
+
+    _NORETURN void Resize(_In_ int width, _In_ int height) noexcept;
 
-        if (bResize)
-        {
-            nDistanceX -= (pRB.x - pLT.x);
-            nDistanceY -= (pRB.y - pLT.y);
-        }
-    }
-
-    void Resize(int width, int height)
-    {
-        int xPos, yPos;
-
-        switch (nAlignType)
-        {
-            case ResizeContent::ALIGN_RIGHT:
-                xPos = width - nDistanceX - (bResize ? pLT.x : 0);
-                yPos = bResize ? (pRB.y - pLT.x) : pLT.y;
-                break;
-            case ResizeContent::ALIGN_BOTTOM:
-                xPos = bResize ? (pRB.x - pLT.x) : pLT.x;
-                yPos = height - nDistanceY - (bResize ? pLT.y : 0);
-                break;
-            case ResizeContent::ALIGN_BOTTOM_RIGHT:
-                xPos = width - nDistanceX - (bResize ? pLT.x : 0);
-                yPos = height - nDistanceY - (bResize ? pLT.y : 0);
-                break;
-            default:
-                xPos = bResize ? (pRB.x - pLT.x) : pLT.x;
-                yPos = bResize ? (pRB.y - pLT.x) : pLT.y;
-                break;
-        }
-
-        if (!bResize)
-            SetWindowPos(hwnd, nullptr, xPos, yPos, 0, 0, SWP_NOSIZE | SWP_NOZORDER);
-        else
-            SetWindowPos(hwnd, nullptr, 0, 0, xPos, yPos, SWP_NOMOVE | SWP_NOZORDER);
-    }
+private:
+    // So we don't have to define a destructor, plus if it doesn't use default_delete we have initialize it explicitly
+    std::unique_ptr<HWND__, decltype(&::DestroyWindow)> hwnd{ nullptr, &::DestroyWindow };
+    POINT pLT = POINT{};
```

Comment:
Jumped the gun a little, there's a simpler version in one my branches (not in a PR) but would rather have destruction be handled more efficiently automatically.